drivers/adc/adc_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2022 Florin Stancu <niflostancu@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc13xx_cc26xx_adc
+
+#include <errno.h>
+
+#define LOG_LEVEL CONFIG_ADC_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(adc_cc13xx_cc26xx);
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/adc.h>
+#include <soc.h>
+
+/* Driverlib includes */
+#include <inc/hw_types.h>
+#include <driverlib/interrupt.h>
+#include <driverlib/ioc.h>
+#include <driverlib/rom.h>
+#include <driverlib/prcm.h>
+#include <driverlib/aux_adc.h>
+#include <ti/devices/cc13x2_cc26x2/inc/hw_aux_evctl.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+
+/**
+ * Channels are based on ADC_COMPB_IN_* hal_ti definitions, max. index is 16 (included).
+ */
+#define CHAN_COUNT 0x10
+
+/** Internal sample time unit conversion entry. */
+struct adc_cc13xx_cc26xx_sample_time_entry {
+	uint16_t time_us;
+	uint8_t reg_value;
+};
+
+/** Maps standard unit sample times (us) to internal (raw hal_ti register) values */
+static const struct adc_cc13xx_cc26xx_sample_time_entry adc_cc13xx_sample_times[] = {
+	{ 3, AUXADC_SAMPLE_TIME_2P7_US },
+	{ 6, AUXADC_SAMPLE_TIME_5P3_US },
+	{ 11, AUXADC_SAMPLE_TIME_10P6_US },
+	{ 22, AUXADC_SAMPLE_TIME_21P3_US },
+	{ 44, AUXADC_SAMPLE_TIME_42P6_US },
+	{ 86, AUXADC_SAMPLE_TIME_85P3_US },
+	{ 170, AUXADC_SAMPLE_TIME_170_US },
```

Comment:
Since you are using `(adc_cc13xx_sample_times[i].time_us > sample_time_us)` in the loop below, this should be 171 (and similarly for the next 6 entries). But it actually seems that using `>=` in the condition and floor rounding of the values would be clearer.